chinese/edit.py

```diff
@@ -27,57 +29,80 @@
 class EditManager:
     def __init__(self):
         addHook('setupEditorButtons', self.setupButton)
-        addHook('loadNote', self.updateButton)
+        addHook('loadNote', self.on_load_note)
         addHook('editFocusLost', self.onFocusLost)
+        gui_hooks.editor_state_did_change.append(self.on_editor_state_did_change)
 
-    def setupButton(self, buttons, editor):
-        self.editor = editor
-        self.buttonOn = False
-        editor._links['chineseSupport'] = self.onToggle
-
-        button = editor._addButton(
+    def setupButton(self, buttons: list[str], editor: aqt.editor.Editor):
+        button = editor.addButton(
             icon=None,
             cmd='chineseSupport',
+            func=self.onToggle,
             tip='Chinese Support',
             label='<b>汉字</b>',
             id='chineseSupport',
-            toggleable=True)
+            toggleable=True
+        )
 
         return buttons + [button]
 
-    def onToggle(self, editor):
-        self.buttonOn = not self.buttonOn
+    def is_enabled_for_note(self, note: anki.notes.Note) -> bool:
+        """Check if the plugin is enabled for the given note."""
+        if not (note_type := note.note_type()):
+            return False
+        return str(note_type["id"]) in config["enabledModels"]
 
-        mid = str(editor.note.note_type()['id'])
+    def onToggle(self, editor: aqt.editor.Editor):
+        if not (note := editor.note):
+            return
+        if not (note_type := note.note_type()):
+            return
 
-        if self.buttonOn and mid not in config['enabledModels']:
-            config['enabledModels'].append(mid)
-        elif not self.buttonOn and mid in config['enabledModels']:
+        mid = str(note_type["id"])
+        if self.is_enabled_for_note(note):
             config['enabledModels'].remove(mid)
+        else:
+            config['enabledModels'].append(mid)
 
         config.save()
-
-    def updateButton(self, editor):
-        enabled = str(editor.note.note_type()['id']) in config['enabledModels']
-
-        if (enabled and not self.buttonOn) or (not enabled and self.buttonOn):
-            editor.web.eval('toggleEditorButton(chineseSupport);')
-            self.buttonOn = not self.buttonOn
-
-    def onFocusLost(self, _, note, index):
-        if not self.buttonOn:
-            return False
-
-        allFields = mw.col.models.field_names(note.note_type())
+        self.updateButton(editor)
+
+    def on_editor_state_did_change(
+        self, editor: aqt.editor.Editor, new_state: aqt.editor.EditorState, old_state: aqt.editor.EditorState
```

Comment:
I think new_state is unused?

Reply:
Yeah it is not used, do you have a linter complaining? If not I would tend towards not `_`ing it in the parameter list.